connect/api/src/main/java/org/apache/kafka/connect/sink/SinkRecord.java

```diff
@@ -57,6 +73,105 @@ public TimestampType timestampType() {
         return timestampType;
     }
 
+    /**
+     * Get the original topic for this sink record, corresponding to the topic of the Kafka record before any
+     * {@link org.apache.kafka.connect.transforms.Transformation Transformation}s were applied. This should be used by
```

Comment:
Nit: formatting/readability
```suggestion
     * {@link org.apache.kafka.connect.transforms.Transformation transformations} were applied. This should be used by
```

(this applies to a bunch of places in this PR; only noting it here to reduce noise)